benchmarking/compute_probability_of_improvement.py

```diff
@@ -0,0 +1,251 @@
+        args.baseline_runs_dir = args.runs_dir
+
+    runs_by_algo_and_env = group_runs_by_algo_and_env(
+        args.runs_dir,
+        only_completed_runs=True,
+    )
+    baseline_runs_by_algo_and_env = group_runs_by_algo_and_env(
+        args.baseline_runs_dir,
+        only_completed_runs=True,
+    )
+
+    algos = sorted(runs_by_algo_and_env.keys())
+    baseline_algos = sorted(baseline_runs_by_algo_and_env.keys())
+
+    try:
+        if len(algos) == 0:
+            raise ValueError(f"The run directory [{args.runs_dir}] contains no runs.")
+
+        if len(baseline_algos) == 0:
+            raise ValueError(
+                f"The baseline run directory [{args.baseline_runs_dir}] "
+                f"contains no runs.",
+            )
+
+        if "algo" not in args is None:
+            if len(algos) == 1:
+                args.algo = algos[0]
+            else:
+                raise ValueError(
+                    f"The run directory [{args.runs_dir}] contains runs for the "
+                    f"algorithms [{', '.join(algos)}]. Please use the --algo option "
+                    f" to specify which algorithms runs to compare.",
+                )
+
+        if args.baseline_algo is None:
+            if len(baseline_algos) == 1:
+                args.baseline_algo = baseline_algos[0]
+            elif args.algo in baseline_algos:
+                args.baseline_algo = args.algo
+            else:
+                raise ValueError(
+                    f"The baseline run directory [{args.baseline_runs_dir}] contains "
+                    f"runs for the algorithms [{', '.join(baseline_algos)}]. "
+                    f"Please use the --baseline-algo option specify which one to "
+                    f"compare to.",
+                )
+
+        if args.algo not in algos:
+            raise ValueError(
+                f"The run directory [{args.runs_dir}] contains runs for the algorithms"
+                f" [{', '.join(algos)}]. You specified [{args.algo}], for which no"
+                f" runs can be found in the run directory",
+            )
+
+        if args.baseline_algo not in baseline_algos:
+            raise ValueError(
+                f"The baseline run directory [{args.baseline_runs_dir}] contains runs "
+                f"for the algorithms [{', '.join(baseline_algos)}]. "
+                f"You specified [{args.baseline_algo}], for which no runs can be found"
+                f" in the baseline run directory",
+            )
+
+        if (args.algo == args.baseline_algo) and (
+            args.runs_dir == args.baseline_runs_dir
+        ):
+            warnings.warn(
+                "You are comparing two equal sets of runs. "
+                "This is probably not what you want.",
+            )
+
+        envs = runs_by_algo_and_env[args.algo].keys()
+        baseline_envs = baseline_runs_by_algo_and_env[args.baseline_algo].keys()
+
+        comparison_envs = set(envs).intersection(set(baseline_envs))
```

Comment:
Similar comment here about `&` (I don't feel strongly about this but IMO it's a bit cleaner.